scvi/external/cell2location/_module.py

```diff
@@ -0,0 +1,336 @@
+
+    def __repr__(self):
+        fmt_string = self.__class__.__name__[1:]
+        fmt_string += '(lower_bound={})'.format(self.lower_bound)
+        return fmt_string
+
+@biject_to.register(_Positive)
+@transform_to.register(_Positive)
+def _transform_to_positive(constraint):
+    return SoftplusTransform()
+
+class ExpToSoftplus(pyro.poutine.messenger.Messenger):
+    
+    def __init__(self):
+        
+        super().__init__()
+    
+    def _process_message(self, msg):
+        if msg["type"] == "param" and msg["name"].endswith("_scale"):
+            msg["kwargs"]["constraint"] = _Positive()
+            
+_handler_name, ExpToSoftplusHandler = pyro.poutine.handlers._make_handler(ExpToSoftplus)
+
+# Define helper gamma distribution
+def Gamma(mu=None, sigma=None, alpha=None, beta=None, shape=None):
+    r"""
+    Function that converts mu/sigma Gamma distribution parametrisation into alpha/beta and
+    returns pyro Gamma distribution with an event of a given shape.
+    A thin wrapper over pyro.dist.Gamma.
+
+    :param mu: mean of Gamma distribution
+    :param sigma: variance of Gamma distribution. Note!!! this is :math:`\sigma^2` not :math:`\sigma`
+    :param alpha: shape parameter of Gamma distribution (mu ** 2 / sigma)
+    :param beta: rate parameter of Gamma distribution (mu / sigma)
+    :param shape: shape of the event / resulting variable. When None the shape is guessed based on input parameters
+
+    :return: pyro Gamma distribution class
+    """
+    if alpha is not None and beta is not None:
+        pass
+    elif mu is not None and sigma is not None:
+        alpha = mu ** 2 / sigma
+        beta = mu / sigma
+    else:
+        raise ValueError('Define (mu and sigma_sq) or (alpha and beta).')
+
+    if shape is None:
+        alpha = torch.tensor(alpha)
+        beta = torch.tensor(beta)
+    else:
+        alpha = torch.ones(shape) * torch.tensor(alpha)
+        beta = torch.ones(shape) * torch.tensor(beta)
+    return dist.Gamma(alpha, beta)
+
+class LocationModelLinearDependentWMultiExperiment(PyroBaseModuleClass):
+    
+    def __init__(self, n_obs, n_var, n_fact, n_exper, batch_size,
+                 cell_state_mat, n_comb: int = 50,
+                 m_g_gene_level_prior={'mean': 1 / 2, 'sd': 1 / 4},
+                 m_g_gene_level_var_prior={'mean_var_ratio': 1},
+                 cell_number_prior={'N_cells_per_location': 8,
+                               'A_factors_per_location': 7,
+                               'Y_combs_per_location': 2.5},
+                 cell_number_var_prior={'N_cells_mean_var_ratio': 1,
+                                   'A_factors_mean_var_ratio': 1,
+                                   'Y_combs_mean_var_ratio': 1},
+                 alpha_g_phi_hyp_prior={'mean': 3, 'sd': 1},
+                 gene_add_alpha_hyp_prior={'mean': 3, 'sd': 1},
+                 gene_add_mean_hyp_prior={'alpha': 1, 'beta': 100},
+                 w_sf_mean_var_ratio=5):
```

Comment:
IMO it's easier to deal with the code if you're explicit about all the hyperparams (each gets it's own parameters).

you should also set your IDE to use the [black formatter](https://black.readthedocs.io/en/stable/) on save. This is mandatory for code in our codebase and is really a painless thing to do.

Reply:
Do you mean a keyword argument for every prior value?

Which IDE do you use, VSCode? I used PyCharm but it breaks my RAM and stopped using it last month (it's just jupyter notebook at the moment).

Reply:
> Do you mean a keyword argument for every prior value?

Yes. And we use VSCode. We might soon also offer a VSCode docker dev container so everyone basically has the same vscode settings.